python-backend/math_utils.py

```diff
@@ -0,0 +1,36 @@
+import sympy
+from sympy import Symbol
+
+import constants
+
+GRAPHABLE_TYPES = (int, float, sympy.core.numbers.Integer, sympy.core.numbers.Float)
+PRECISION = constants.PRECISION
+X_VALUES = range(0, 1000, 25)
+
+
+def error_coordinates(expression: sympy.core, symbol: Symbol, limit: sympy.core.numbers):
+    """
+    graph coords for the error of an expression log(|expression - L|) in terms of symbol
+    :param expression: univariate mathematical expression comprising a ratio of two polynomials
+    :param symbol: the variable used in expression
+    :param limit: the computed limit of the expression as it goes to infinity
+    :return: array of [x,y] pairs for graphing purposes
+    """
+    return [[val, float(sympy.log(abs(expression.subs(symbol, val) - limit)).evalf(PRECISION))] for val in X_VALUES if
```

Comment:
This return value is pretty complicated, so I tried to add indentation to make it more legible but I'm not sure that I got it right.

```
[
    [
        val, 
        float(sympy.log(abs(expression.subs(symbol, val) - limit)).evalf(PRECISION))
    ]
    for val in X_VALUES 
    if type(sympy.log(abs(expression.subs(symbol, val) - limit)).evalf(PRECISION)) in GRAPHABLE_TYPES
]
```

Consider creating a `dataclass`-based type to replace the inner list.

Reply:
I agree that it's illegible. It's just a list of `int, float` pairs so I specified that and broke out the nested logic for legibility.